### PR TITLE
Move async example to asynchronous version of proc

### PR DESCRIPTION
**Pull request by vercingetorx**

### CODE REVIEW

**Review by dom96 — APPROVED** [review]
Looks good. For bonus points, can we get the same example for synchronous httpclient but one that is sync IO?

**Comment by vercingetorx** [comment]
I copied the example verbatim, removing the async elements. But in hindsight, maybe the examples should use result instead of return as the Nim documentation suggests using return only for control flow. Just a minor nitpick.

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -616,6 +616,17 @@ proc newAsyncHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
...
+  runnableExamples:
+    import std/[asyncdispatch, httpclient, strutils]
+
+    proc asyncProc(): Future[string] {.async.} =
+      var client = newAsyncHttpClient()
+      return await client.getContent("http://example.com")
+
+    let exampleHtml = waitFor asyncProc()
+    assert "Example Domain" in exampleHtml
+    assert not ("Pizza" in exampleHtml)
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]:
```suggestion
    assert "Pizza" notin exampleHtml
```
(+ fix original example too)

File: lib/pure/httpclient.nim [hunk 2]
```diff
@@ -616,6 +616,17 @@ proc newAsyncHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
   ## connections.
   ##
   ## `headers` specifies the HTTP Headers.
+  runnableExamples:
+    import std/[asyncdispatch, httpclient, strutils]
+
+    proc asyncProc(): Future[string] {.async.} =
+      var client = newAsyncHttpClient()
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]:
```suggestion
      let client = newAsyncHttpClient()
```
(ditto with original example)

File: lib/pure/httpclient.nim [hunk 3]
```diff
@@ -574,13 +574,13 @@ proc newHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
...
   runnableExamples:
-    import std/[asyncdispatch, httpclient, strutils]
+    import std/[httpclient, strutils]
 
-    proc asyncProc(): Future[string] {.async.} =
-      var client = newAsyncHttpClient()
-      return await client.getContent("http://example.com")
+    proc syncProc(): string =
+      var client = newHttpClient()
+      return client.getContent("http://example.com")
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]:
```suggestion
      result = client.getContent("http://example.com")
```

ditto with other case

File: lib/pure/httpclient.nim [hunk 4]
```diff
@@ -574,13 +574,13 @@ proc newHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
...
   ## `headers` specifies the HTTP Headers.
   runnableExamples:
-    import std/[asyncdispatch, httpclient, strutils]
+    import std/[httpclient, strutils]
 
-    proc asyncProc(): Future[string] {.async.} =
-      var client = newAsyncHttpClient()
-      return await client.getContent("http://example.com")
+    proc syncProc(): string =
+      var client = newHttpClient()
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]: let (did you see my previous comments?)

File: lib/pure/httpclient.nim [hunk 5]
```diff
@@ -574,13 +574,13 @@ proc newHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
...
+      result = client.getContent("http://example.com")
 
-    let exampleHtml = waitFor asyncProc()
+    let exampleHtml = syncProc()
     assert "Example Domain" in exampleHtml
     assert not ("Pizza" in exampleHtml)
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]: notin

File: lib/pure/httpclient.nim [hunk 6]
```diff
@@ -574,13 +574,13 @@ proc newHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
   ##
   ## `headers` specifies the HTTP Headers.
   runnableExamples:
-    import std/[asyncdispatch, httpclient, strutils]
+    import std/[httpclient, strutils]
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]:
```suggestion
    import std/strutils
```
(current module doesn't need to be specified in runnableExamples)

File: lib/pure/httpclient.nim [hunk 7]
```diff
@@ -574,13 +574,13 @@ proc newHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
...
 
-    proc asyncProc(): Future[string] {.async.} =
-      var client = newAsyncHttpClient()
-      return await client.getContent("http://example.com")
+    proc syncProc(): string =
+      var client = newHttpClient()
+      result = client.getContent("http://example.com")
 
-    let exampleHtml = waitFor asyncProc()
+    let exampleHtml = syncProc()
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]: actually you don't need `syncProc`, just use this directly, simpler is better:
```nim
let exampleHtml = newHttpClient().getContent("http://example.com")
```

File: lib/pure/httpclient.nim [hunk 8]
```diff
@@ -616,6 +616,17 @@ proc newAsyncHttpClient*(userAgent = defUserAgent, maxRedirects = 5,
   ## connections.
   ##
   ## `headers` specifies the HTTP Headers.
+  runnableExamples:
+    import std/[asyncdispatch, httpclient, strutils]
```
Comment (timotheecour) [inline lib/pure/httpclient.nim]:
```suggestion
    import std/[asyncdispatch, strutils]
```
(same reason as above)

File: lib/pure/httpclient.nim (hunk 4 above)
Comment (vercingetorx) [reply lib/pure/httpclient.nim]: yeah I see them now. will fix.

**Review by timotheecour — APPROVED** [review]

**Comment by timotheecour** [comment]
thanks! looking forward for more PRs :)

**Pull request merged**